### PR TITLE
Added LSTM BF16 and fixed GRU BF16

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
New features

### PR changes
OPs

### Describe
Added support for LSTM bf16. Added support for bf16 and fp32 weights and h0 memory datatypes. In future int8 will be supported as weights datatype too.

### CODE REVIEW

**Comment by jakpiase** [comment]
@jczaja @wojtuss Could you please take a look?

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@luotao1 Could you please start your review?

**Review by luotao1 — APPROVED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/mkldnn/test_fusion_gru_bf16_mkldnn_op.py [hunk 1]
```diff
@@ -74,20 +83,30 @@ def setUp(self):
...
+                'WeightH': wh_fp32
+            }
 
         if self.with_bias:
             self.inputs['Bias'] = bias
 
         if self.with_h0:
-            self.inputs['H0'] = h0_bf16
+            if self.weights_dtype == 'bf16':
+                self.inputs['H0'] = h0_bf16
```
Comment (lelelelelez) [inline python/paddle/fluid/tests/unittests/mkldnn/test_fusion_gru_bf16_mkldnn_op.py:104]: hi @jakpiase   , you use variable `h0_bf16` before assignment

**Pull request merged**